calyx-backend/src/firrtl.rs

```diff
@@ -34,14 +35,35 @@ impl Backend for FirrtlBackend {
 
     fn emit(ctx: &ir::Context, file: &mut OutputFile) -> CalyxResult<()> {
         let out = &mut file.get_write();
-        let mut top_level_component = String::from("main");
-        // Quick pass to check whether there exists a top-level component that we should replace main with.
+        writeln!(out, "circuit {}:", ctx.entrypoint)?;
+        // Pass to output any necessary extmodule statements (for primitive calls)
+        let mut extmodule_set: HashSet<String> = HashSet::new();
         for comp in ctx.components.iter() {
-            if comp.attributes.has(ir::BoolAttr::TopLevel) {
-                top_level_component = comp.name.to_string().clone();
+            for cell in comp.cells.iter() {
+                let cell_borrowed = cell.as_ref().borrow();
+                if cell_borrowed.type_name().is_some() {
+                    if let ir::CellType::Primitive {
+                        name,
+                        param_binding,
+                        is_comb: _,
+                        latency: _,
```

Comment:
nit: if you don't need these, you should be able to replace the empty matches with `..`.

https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#ignoring-remaining-parts-of-a-value-with-